src/utils.js

```diff
@@ -8,3 +8,15 @@ export function circleCircleCollision(circle1, circle2) {
     (circle1.size + circle2.size) ** 2
   )
 }
+
+/**
+ * algorithm for 2D baddy -> player collision
+ * @see https://developer.mozilla.org/en-US/docs/Games/Techniques/2D_collision_detection */
+export function baddyPlayerCollision(baddy, player) {
```

Comment:
We can kill two birds with one stone if we treat the player as a circle. That lets us use the `circleCircleCollision` function and also makes the player's corners not trigger collision so they have a smaller hit box than the full rectangle. Just need to give the player a `size` property, probably equal to it's width.

Reply:
Ah ya that's a lot better